From 180d858c6b7dfb90c8b3dd5aaafaa384118a5cc9 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 25 Aug 2014 07:53:06 -0400
Subject: [tools] perf: Fix AAAAARGH64 memory barriers

Message-id: <1408953210-26343-299-git-send-email-jolsa@redhat.com>
Patchwork-id: 88933
O-Subject: [PATCH RHEL7.1 BZ1133083 298/322] perf tools: Fix AAAAARGH64 memory barriers
Bugzilla: 1133083
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1133083
https://bugzilla.redhat.com/show_bug.cgi?id=1133083

upstream
========
commit f428ebd184c82a7914b2aa7e9f868918aaf7ea78
Author: Peter Zijlstra <peterz@infradead.org>
Date: Fri Jan 24 16:40:02 2014 +0100

description
===========
Someone got the load and store barriers mixed up for AAAAARGH64.  Turn
them the right side up.

Fixes: a94d342b9cb0 ("tools/perf: Add required memory barriers")
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/perf.h b/tools/perf/perf.h
index 7daa806..e84fa26 100644
--- a/tools/perf/perf.h
+++ b/tools/perf/perf.h
@@ -100,8 +100,8 @@
 
 #ifdef __aarch64__
 #define mb()  asm volatile("dmb ish" ::: "memory")
-#define wmb()  asm volatile("dmb ishld" ::: "memory")
-#define rmb()  asm volatile("dmb ishst" ::: "memory")
+#define wmb()  asm volatile("dmb ishst" ::: "memory")
+#define rmb()  asm volatile("dmb ishld" ::: "memory")
 #define cpu_relax() asm volatile("yield" ::: "memory")
 #endif
 
-- 
1.7.1